### PR TITLE
LPS-195040 Unable to import custom object with multiselect picklist field because of ClassCastException

### PR DESCRIPTION
**Pull request by matijapetanjek**

### CODE REVIEW

**Comment by matijapetanjek** [comment]
ci:test:relevant

**Review by ccorreagg — COMMENTED** [review]
LGTM! However, when I test it locally, I always get the following exception:

```
2023-09-18 12:45:27.765 WARN  [default-21][ObjectEntryModelDocumentContributor:65] Unable to index object entry 35231
com.liferay.portal.kernel.exception.SystemException: java.lang.IllegalStateException: EntityManagerFactory is closed
	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.processException(BasePersistenceImpl.java:640) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.dslQuery(BasePersistenceImpl.java:308) ~[portal-kernel.jar:?]
	at com.liferay.object.service.impl.ObjectEntryLocalServiceImpl._list(ObjectEntryLocalServiceImpl.java:3352) ~[?:?]
	at com.liferay.object.service.impl.ObjectEntryLocalServiceImpl.getValues(ObjectEntryLocalServiceImpl.java:1044) ~[?:?]
	at com.liferay.object.service.impl.ObjectEntryLocalServiceImpl.getValues(ObjectEntryLocalServiceImpl.java:1017) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_382]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_382]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_382]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_382]
	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:41) ~[portal-impl.jar:?]
	at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:60) ~[portal-impl.jar:?]
	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:48) ~[portal-impl.jar:?]
	at com.liferay.change.tracking.internal.aop.CTTransactionAdvice.invoke(CTTransactionAdvice.java:70) ~[?:?]
	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:48) ~[portal-impl.jar:?]
	at com.liferay.portal.spring.aop.AopInvocationHandler.invoke(AopInvocationHandler.java:40) ~[portal-impl.jar:?]
	at com.sun.proxy.$Proxy755.getValues(Unknown Source) ~[?:?]
	at com.liferay.object.internal.search.spi.model.index.contributor.ObjectEntryModelDocumentContributor._contribute(ObjectEntryModelDocumentContributor.java:131) ~[?:?]
	at com.liferay.object.internal.search.spi.model.index.contributor.ObjectEntryModelDocumentContributor.contribute(ObjectEntryModelDocumentContributor.java:61) ~[?:?]
	at com.liferay.object.internal.search.spi.model.index.contributor.ObjectEntryModelDocumentContributor.contribute(ObjectEntryModelDocumentContributor.java:43) ~[?:?]
	at com.liferay.portal.search.internal.indexer.IndexerDocumentBuilderImpl.lambda$getDocument$1(IndexerDocumentBuilderImpl.java:54) ~[?:?]
	at com.liferay.osgi.service.tracker.collections.internal.list.ServiceTrackerListImpl.lambda$forEach$0(ServiceTrackerListImpl.java:73) ~[com.liferay.osgi.service.tracker.collections.jar:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_382]
	at com.liferay.osgi.service.tracker.collections.internal.list.ServiceTrackerListImpl.forEach(ServiceTrackerListImpl.java:72) ~[com.liferay.osgi.service.tracker.collections.jar:?]
	at com.liferay.portal.search.internal.indexer.IndexerDocumentBuilderImpl.getDocument(IndexerDocumentBuilderImpl.java:52) ~[?:?]
	at com.liferay.portal.search.internal.indexer.IndexerWriterImpl.reindex(IndexerWriterImpl.java:220) ~[?:?]
	at com.liferay.portal.search.internal.indexer.IndexerWriterImpl.reindex(IndexerWriterImpl.java:206) ~[?:?]
	at com.liferay.portal.search.internal.indexer.IndexerWriterImpl.reindex(IndexerWriterImpl.java:156) ~[?:?]
	at com.liferay.portal.search.internal.indexer.DefaultIndexer.reindex(DefaultIndexer.java:250) ~[?:?]
	at com.liferay.portal.search.internal.buffer.NoAutoCommitIndexer.reindex(NoAutoCommitIndexer.java:210) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor705.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_382]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_382]
	at com.liferay.portal.search.internal.buffer.IndexerRequest.execute(IndexerRequest.java:81) ~[?:?]
	at com.liferay.portal.search.internal.buffer.IndexerRequestBufferExecutor.execute(IndexerRequestBufferExecutor.java:61) ~[?:?]
	at com.liferay.portal.search.internal.buffer.IndexerRequestBufferExecutor.execute(IndexerRequestBufferExecutor.java:30) ~[?:?]
	at com.liferay.portal.search.internal.buffer.IndexerRequestBufferTransactionLifecycleListener.committed(IndexerRequestBufferTransactionLifecycleListener.java:31) ~[?:?]
	at com.liferay.portal.kernel.transaction.TransactionLifecycleNotifier.fireTransactionCommittedEvent(TransactionLifecycleNotifier.java:56) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.transaction.TransactionLifecycleNotifier$1.doCommitted(TransactionLifecycleNotifier.java:25) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.transaction.NewTransactionLifecycleListener.committed(NewTransactionLifecycleListener.java:20) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.transaction.TransactionLifecycleManager.fireTransactionCommittedEvent(TransactionLifecycleManager.java:25) ~[portal-kernel.jar:?]
	at com.liferay.portal.spring.transaction.DefaultTransactionExecutor.commit(DefaultTransactionExecutor.java:42) ~[portal-impl.jar:?]
	at com.liferay.portal.spring.transaction.BaseTransactionExecutor.execute(BaseTransactionExecutor.java:35) ~[portal-impl.jar:?]
	at com.liferay.portal.spring.transaction.TransactionInvokerImpl.invoke(TransactionInvokerImpl.java:30) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.transaction.TransactionInvokerUtil.invoke(TransactionInvokerUtil.java:19) ~[portal-kernel.jar:?]
	at com.liferay.batch.engine.internal.BatchEngineImportTaskExecutorImpl._commitItems(BatchEngineImportTaskExecutorImpl.java:157) ~[?:?]
	at com.liferay.batch.engine.internal.BatchEngineImportTaskExecutorImpl._importItems(BatchEngineImportTaskExecutorImpl.java:331) ~[?:?]
	at com.liferay.batch.engine.internal.BatchEngineImportTaskExecutorImpl.lambda$execute$0(BatchEngineImportTaskExecutorImpl.java:106) ~[?:?]
	at com.liferay.batch.engine.internal.BatchEngineTaskExecutorUtil.execute(BatchEngineTaskExecutorUtil.java:44) ~[?:?]
	at com.liferay.batch.engine.internal.BatchEngineImportTaskExecutorImpl.execute(BatchEngineImportTaskExecutorImpl.java:104) ~[?:?]
	at com.liferay.batch.engine.internal.BatchEngineImportTaskExecutorImpl.execute(BatchEngineImportTaskExecutorImpl.java:74) ~[?:?]
	at com.liferay.headless.batch.engine.internal.resource.v1_0.ImportTaskResourceImpl.lambda$_importFile$2(ImportTaskResourceImpl.java:476) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_382]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266) [?:1.8.0_382]
	at java.util.concurrent.FutureTask.run(FutureTask.java) [?:1.8.0_382]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_382]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_382]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_382]
Caused by: java.lang.IllegalStateException: EntityManagerFactory is closed
	at org.hibernate.internal.SessionFactoryImpl.validateNotClosed(SessionFactoryImpl.java:547) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionFactoryImpl.getCache(SessionFactoryImpl.java:864) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.AbstractSharedSessionContract.<init>(AbstractSharedSessionContract.java:159) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.AbstractSessionImpl.<init>(AbstractSessionImpl.java:29) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionImpl.<init>(SessionImpl.java:230) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionFactoryImpl$SessionBuilderImpl.openSession(SessionFactoryImpl.java:1334) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionFactoryImpl.openSession(SessionFactoryImpl.java:486) ~[hibernate-core.jar:5.6.7.Final]
	at com.liferay.portal.spring.hibernate.PortalCurrentSessionContext.currentSession(PortalCurrentSessionContext.java:67) ~[portal-impl.jar:?]
	at org.hibernate.internal.SessionFactoryImpl.getCurrentSession(SessionFactoryImpl.java:508) ~[hibernate-core.jar:5.6.7.Final]
	at com.liferay.portal.dao.orm.hibernate.SessionFactoryImpl.openSession(SessionFactoryImpl.java:87) ~[portal-impl.jar:?]
	at com.liferay.portal.dao.orm.hibernate.VerifySessionFactoryWrapper.openSession(VerifySessionFactoryWrapper.java:93) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.openSession(BasePersistenceImpl.java:628) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.dslQuery(BasePersistenceImpl.java:233) ~[portal-kernel.jar:?]
	... 55 more
```

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/MultiselectPicklistObjectFieldBusinessType.java [hunk 1]
```diff
@@ -100,6 +102,29 @@ public PropertyDefinition.PropertyType getPropertyType() {
 		return PropertyDefinition.PropertyType.TEXT;
 	}
 
+	@Override
+	public Object getValue(
```
Comment (ccorreagg) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/MultiselectPicklistObjectFieldBusinessType.java:106]: I miss a test where we are using this method, not sure if a unit test or an integration test (for example, the REST API as I guess the POST should work the same way)

Comment (matijapetanjek) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/MultiselectPicklistObjectFieldBusinessType.java:106]: added a test, run it with gw testIntegration --tests MultiselectPicklistObjectFieldBusinessTypeTest

**Review by ccorreagg — APPROVED** [review]

**Comment by matijapetanjek** [comment]
ci:forward:force

**Comment by matijapetanjek** [comment]
ci:stop

**Comment by matijapetanjek** [comment]
ci:test:sf

**Comment by matijapetanjek** [comment]
ci:forward

**Comment by matijapetanjek** [comment]
ci:forward:force

**Pull request closed without merging**